### ISSUE TITLE
Feat: Object Detection from Video & Statistical Analysis

### ISSUE THREAD

**Issue by Jibesh10101011** [AOSSIE-Org/PictoPy#274]
## Description 

This feature will enable object detection from videos and provide statistical insights about detected objects.

## Feature Details 

1. Users will have an option in the "Videos" section to process videos.
2. While processing, users can watch the video with detected objects in real-time. 
3. After processing, a statistical distribution of detected objects (e.g., {person: 4, animal: 5}) will be provided. e.g. 

![Image](https://github.com/user-attachments/assets/4678ec35-6959-45ac-a2f8-042559602e97)

4. Option to save metadata related to the statistical distribution for future reference.

## Implementation Plan

- [x] Integrate the YOLOv8 model with the existing object detection architecture.
- [x] Implement WebSockets to send base64-encoded frames to the frontend for real-time visualization.
- [x] Frame transmission via WebSocket may introduce some latency.
- [x] Use Matplotlib for statistical analysis and visualization of detected objects.

  

## Demo Video In testing Phase

https://github.com/user-attachments/assets/10c56cc7-f7c1-4ab0-9e32-dab73246d2a7

### Record

- [x] I agree to follow this project's Code of Conduct
- [x] I want to work on this issue

**Comment by Jibesh10101011** [comment]
@Pranav0-0Aggarwal I am working on this feature , can you assign this to me ?

**Comment by Mansi08git** [comment]
I have worked on projects similar like this. In one of my project, I worked on face-detection and face recognition. Please assign this issue to me

**Comment by Jibesh10101011** [comment]
@Mansi08git I have completed my work for this issue . you may find another

**Comment by Ashutosh-KARNX7** [comment]
@Jibesh10101011 what is the need of this

**Comment by arpanhub** [comment]
@Jibesh10101011 have you completed this issue, or is it still in the testing phase?
If it's in testing, how many epochs did it take to achieve that accuracy percentage, and what learning rate are you using?

**Comment by Jibesh10101011** [comment]
> [@Jibesh10101011](https://github.com/Jibesh10101011) have you completed this issue, or is it still in the testing phase? If it's in testing, how many epochs did it take to achieve that accuracy percentage, and what learning rate are you using?

@arpanhub My task for this issue is completed. I am not training any model here , the YOLO model is already provided, and I have simply utilized its functionality. I am waiting for @Pranav0-0Aggarwal to review this issue. After that, I will raise a PR.

**Comment by Ashutoshx7** [comment]
@Jibesh10101011 I working on something like camera face detection I think we should go for the pr

**Comment by Jibesh10101011** [comment]
> [@Jibesh10101011](https://github.com/Jibesh10101011) I working on something like camera face detection I think we should go for the pr

Yes @Ashutosh-KARNX7 I think you are right , but needs our mentor @Pranav0-0Aggarwal 's review once for this issue .

**Comment by Ashutoshx7** [comment]
Bro this is a usefull feature he is busy person (I think he is a final year student ) I don' t think so he has now that much time that he can do this.  So go for the pr

**Comment by aditi25srivastava** [comment]
@Jibesh10101011 Heyy!!!!! I would love to work on this issue. Could you please assign it to me ?

**Comment by Jibesh10101011** [comment]
> [@Jibesh10101011](https://github.com/Jibesh10101011) Heyy!!!!! I would love to work on this issue. Could you please assign it to me ?

Hey @aditi25srivastava ! Sure, you can work on this
Please ask @rahulharpal1603 to assign the issue to you — he’s the current maintainer of Pictopy.

**Comment by aditi25srivastava** [comment]
@rahulharpal1603 Heyy!!!!! I would love to work on this issue. Could you please assign it to me ?
